analyzers/tests/SonarAnalyzer.TestFramework/Common/SourceGeneratorProvider.cs

```diff
@@ -26,13 +26,34 @@ namespace SonarAnalyzer.TestFramework.Common;
 public static class SourceGeneratorProvider
 {
     private static readonly string RazorSourceGeneratorPath =
-        Path.Combine(Path.GetDirectoryName(typeof(SourceGeneratorProvider).Assembly.Location), "Dependencies", "Microsoft.CodeAnalysis.Razor.Compiler.SourceGenerators.dll");
+        Path.Combine(LatestSdkFolder(), "Sdks", "Microsoft.NET.Sdk.Razor", "source-generators", "Microsoft.CodeAnalysis.Razor.Compiler.SourceGenerators.dll");
 
     public static AnalyzerFileReference[] SourceGenerators { get; } =
     [
-        new(RazorSourceGeneratorPath, new AssemblyLoader())
+        new(CheckRazorSourceGeneratorPath(), new AssemblyLoader())
     ];
 
+    public static string CheckRazorSourceGeneratorPath() =>
+        File.Exists(RazorSourceGeneratorPath) ? RazorSourceGeneratorPath : throw new FileNotFoundException($"Razor sourcegenerator not found: {RazorSourceGeneratorPath}");
+
+    public static string LatestSdkFolder()
+    {
+        var objectAssembly = typeof(object).Assembly;
+        var objectAssemblyDirectory = new FileInfo(objectAssembly.Location).Directory; // C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.4
+        var dotnetDirectory = objectAssemblyDirectory.Parent.Parent.Parent;            // C:\Program Files\dotnet
+        var sdkDirectory = Path.Combine(dotnetDirectory.FullName, "sdk");              // C:\Program Files\dotnet\sdk
+        if (!Directory.Exists(sdkDirectory))
+        {
+            throw new NotSupportedException($"Razor analysis is only supported for .Net Core.");
+        }
+        var specificMajorVersionSdkDirectories = Directory.GetDirectories(sdkDirectory, $"{objectAssembly.GetName().Version.Major}.*", SearchOption.TopDirectoryOnly);
+        if (specificMajorVersionSdkDirectories.Length == 0)
+        {
+            throw new DirectoryNotFoundException($"SDK directory not found for version {objectAssembly.GetName().Version.Major}");
+        }
+        return specificMajorVersionSdkDirectories.OrderByDescending(x => Version.Parse(new DirectoryInfo(x).Name)).FirstOrDefault();
```

Comment:
Very nice!
You can use `.First()` here since you checked for empty collection before.